### ISSUE TITLE
what's the "morning" in setting file?

### ISSUE THREAD

**Issue by marsan-ma** [thestr4ng3r/tlou2-guitarbot#1]
Hi, very interesting project!
But the Chiaki source code has so many mysterious var names I cannot get them.

Like in this `tlou2-guitarbot` project, the read_settings() in main.c needs a setting file, 
but I'm not sure what's the 1st line "host" and the 3rd line "morning" in base64 should be.

Could you provide a dummy setting example so we could learn what's that and what's the requested format? 
Thanks!

**Comment by thestr4ng3r** [comment]
The regist key and morning are explained in here: https://github.com/thestr4ng3r/chiaki/issues/345
You can register using the Chiaki GUI and then extract the values as described there.

An example file could look like this:
```
192.168.13.37
0123abcd
aaaaaaaaaaaaaaaaaaaaaa==
1
30
```

**Comment by marsan-ma** [comment]
cool, thanks! 

oh wait, I got a stupid question. I'm using Mac and seems like the Chiaki config file is not under `~/.config/Chiaki/Chiaki.conf`. 

I tried to `sudo find / -name Chiaki.conf` but didn't got anything. Do you have idea where the path might be? thanks!

**Comment by thestr4ng3r** [comment]
morning comes from registration with the console, not PSN OAuth. On macOS it's in `~/Library/Preferences/com.chiaki.Chiaki.plist` but it's a binary plist, so you have to open it in Xcode or something to extract the info. The key names are the same as in the Linux variant.

**Comment by marsan-ma** [comment]
thanks! now it passes the settings checking part!
but it quit right after start, the error message is pretty hard to find where to track ...

> [I] Starting session request
> [I] Trying to request session from 192.168.2.110:9295
> [I] Connected to 192.168.2.110:9295
> [I] Sending session request
> [E] Reported Application Reason: 0x80108bff (unknown)
> [I] Session has quit

I'm wondering maybe I make it / or use it in the wrong way?
I only make in the root directory and run the `guitarbot` binary with only 1 param as the setting file containing the parameters (host, key, morning...) above.

also, will it kick-off chiaki gui by itself? or shall I run chiaki gui first then run guitarbot?
I tried so and it give me almost the same result:

> [I] Starting session request
> [I] Trying to request session from 192.168.2.110:9295
> [I] Connected to 192.168.2.110:9295
> [I] Sending session request
> [E] Reported Application Reason: 0x80108b10 (Remote is already in use)
> [I] Session has quit

**Comment by thestr4ng3r** [comment]
> thanks! now it passes the settings checking part!
> but it quit right after start, the error message is pretty hard to find where to track ...
> 
> > [I] Starting session request
> > [I] Trying to request session from 192.168.2.110:9295
> > [I] Connected to 192.168.2.110:9295
> > [I] Sending session request
> > [E] Reported Application Reason: 0x80108bff (unknown)
> > [I] Session has quit
> 
> I'm wondering maybe I make it / or use it in the wrong way?
> I only make in the root directory and run the `guitarbot` binary with only 1 param as the setting file containing the parameters (host, key, morning...) above.

Either the settings are still wrong or (more likely) your PS4 is on 8.0 already, with which the outdated chiaki in the submodule was not compatible. I've updated the submodule so try pulling master and running `git submodule update`.

> also, will it kick-off chiaki gui by itself? or shall I run chiaki gui first then run guitarbot?
> I tried so and it give me almost the same result:
> 
> > [I] Starting session request
> > [I] Trying to request session from 192.168.2.110:9295
> > [I] Connected to 192.168.2.110:9295
> > [I] Sending session request
> > [E] Reported Application Reason: 0x80108b10 (Remote is already in use)
> > [I] Session has quit

The bot is self-contained, it does not use the gui at all. If you run both, you get exactly the error you see there, which means you can't connect two clients at the same time.

**Comment by marsan-ma** [comment]
mmm .. I pulled the latest update and build again, but still got the same error

> [I] Starting session request
> [I] Trying to request session from 192.168.2.110:9295
> [I] Connected to 192.168.2.110:9295
> [I] Sending session request
> [E] Reported Application Reason: 0x80108bff (unknown)
> [I] Session has quit

I have a stupid question since I'm not familiar with C project: do I have to go to each submodule and build them first?
Currently, I only build on root path `tlou2-guitarbot`, just `cmake .` and `make` it.

**Comment by thestr4ng3r** [comment]
No you don't need to build anything manually, only the top-level guitarbot project. This all looks good so I can only conclude that your settings (morning and regist key) are not yet correct, or not correctly encoded.

**Comment by marsan-ma** [comment]
arh, yeah that might be the case. 

I converted  `~/Library/Preferences/com.chiaki.Chiaki.plist` to XML by `plutil -convert xml1 com.chiaki.Chiaki.plist.plist`, then I found seems like both my `registered_hosts.1.rp_key` and `registered_hosts.1.rp_regist_key` are already in like `abcdABCD12341234==` format.

I guess maybe they're both in base64?  in that case `morning` is just copy paste, but what's the expected `regist_key`?
Thanks for the help!
